irodsadmin/management/commands/irodsorphans.py

```diff
@@ -25,184 +26,224 @@
 table_builder = SampleSheetTableBuilder()
 
 
-def get_assay_collections(assays, irods_backend):
-    """Return a list of all assay collection names."""
-    return [irods_backend.get_path(a) for a in assays]
-
-
-def get_assay_subcollections(studies, irods_backend):
-    """Return a list of all assay row colletion names."""
-    collections = []
-    for study in studies:
-        try:
-            study_tables = table_builder.get_study_tables(study)
-        except Exception as ex:
-            logger.error(
-                'Study table building exception for "{}" '
-                'in project "{}" ({}): {}'.format(
-                    study.get_display_name(),
-                    study.investigation.project.title,
-                    study.investigation.project.sodar_uuid,
-                    ex,
-                )
+# Local constants
+DELETED = '<DELETED>'
+ERROR = '<ERROR>'
+
+
+class Command(BaseCommand):
+    """Command to find orphans in iRODS collections."""
+
+    help = 'Find orphans in iRODS project collections.'
+
+    def __init__(self):
+        super().__init__()
+        self.irods_backend = get_backend_api('omics_irods')
+
+    def _sort_colls_on_projects(self, all_project_collections, project_list):
+        """Helper function to sort collections based on project list"""
+        # Separate strings with UUIDs from those without
+        strings_with_uuids = []
+        strings_without_uuids = []
+
+        # Iterate through L1 and classify strings
+        for coll in all_project_collections:
+            match = re.search(
+                r'[a-f0-9]{8}-([a-f0-9]{4}-){3}[a-f0-9]{12}', coll.path
```

Comment:
Note that simply searching for the UUID is not OK in this case. Nothing is preventing a user from creating a collection to a landing zone with the name of an UUID of some other project. We need to check that the project UUID appears at the correct place in the path, namely after `{projects_path}/{two_first_characters_of_uuid}/{PROJECT_UUID_GOES_HERE}`. This we already too e.g. when validating iRODS delete requests.

One possible simple way would be to call `IrodsAPI.get_path(project)` for each project when we get the project list and store the project path in the sorted list. Then in here we could simply check if the start of the path is found with `any(startswith())` etc. But there might be an even better approach, feel free to give one a try :)